.github/workflows/continuous-integration.yml

```diff
@@ -2,67 +2,111 @@
 env:
-  DEBUG: 1
+  PYTHON_TARGET: 3.8
+  # Django
+  DJANGO_SETTINGS_MODULE: tcf_core.settings.ci
   SECRET_KEY: ${{ secrets.SECRET_KEY }}
+  DEBUG: 1
+  # database
+  DB_NAME: tcf_db  # arbitrary string
+  DB_USER: postgres  # default user
+  DB_PASSWORD: postgres  # default password
+  DB_HOST: localhost  # required for GitHub Actions
+  DB_PORT: 5432  # default port
+  # social-auth-app-django
   SOCIAL_AUTH_GOOGLE_OAUTH2_KEY: ${{ secrets.SOCIAL_AUTH_GOOGLE_OAUTH2_KEY }}
   SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET: ${{ secrets.SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET }}
+  # elasticsearch
   ES_PUBLIC_API_KEY: ${{ secrets.ES_PUBLIC_API_KEY }}
   ES_COURSE_SEARCH_ENDPOINT: ${{ secrets.ES_COURSE_SEARCH_ENDPOINT }}
   ES_INSTRUCTOR_SEARCH_ENDPOINT: ${{ secrets.ES_INSTRUCTOR_SEARCH_ENDPOINT }}
+  ES_COURSE_DOCUMENTS_ENDPOINT: ${{ secrets.ES_COURSE_DOCUMENTS_ENDPOINT }}
+  ES_INSTRUCTOR_DOCUMENTS_ENDPOINT: ${{ secrets.ES_INSTRUCTOR_DOCUMENTS_ENDPOINT }}
+  ES_PRIVATE_API_KEY: ${{ secrets.ES_PRIVATE_API_KEY }}
+  # discord
   DISCORD_URL_BUG: ${{ secrets.DISCORD_URL_BUG }}
   DISCORD_URL_FEEDBACK: ${{ secrets.DISCORD_URL_FEEDBACK }}
-
 jobs:
-  build:
+  pylint:
     runs-on: ubuntu-latest
-    strategy:
-      max-parallel: 4
-      matrix:
-        python-version: [3.8]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ env.PYTHON_TARGET }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.PYTHON_TARGET }}
+
+      - name: Install Python packages, excluding the unnecessary ones
+        run: |
+          python -m pip install --upgrade pip
+          sed -i '/\(tqdm\|gunicorn\|django\-heroku\|uwsgi\)/d' requirements.txt
+          pip install -r requirements.txt
+
+      - name: Run pylint
+        run: pylint --django-settings-module=${{env.DJANGO_SETTINGS_MODULE}} --load-plugins pylint_django tcf_website tcf_core
 
+  django:
+    runs-on: ubuntu-latest
+    outputs:
+      code-coverage: ${{ steps.coverage.outputs.percentage }}
     services:
       postgres:
         image: postgres:12.2
         env:
-          POSTGRES_USER: tcf_django
-          POSTGRES_PASSWORD: s3kr1t
-          POSTGRES_DB: tcf_db
+          POSTGRES_USER: ${{ env.DB_USER}}
+          POSTGRES_PASSWORD: ${{ env.DB_PASSWORD }}
+          POSTGRES_DB: ${{ env.DB_NAME }}
         ports:
           - 5432:5432
         # needed because the postgres container does not provide a healthcheck
         options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
-
     steps:
       - uses: actions/checkout@v2
 
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Set up Python ${{ env.PYTHON_TARGET }}
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: ${{ env.PYTHON_TARGET }}
 
-      - name: Install Python packages
+      - name: Install Python packages, excluding the unnecessary ones
```

Comment:
> By default, a workflow with multiple jobs will run those jobs in parallel. 

https://docs.github.com/en/actions/learn-github-actions/introduction-to-github-actions#jobs
The `django` job isn't _reinstalling_ the packages--it is installing the packages _in a different container_. I wanted to minimize the total build time, although It won't make much of a difference.

I could have combined the two (or three) jobs into one, but in that case, we might need to wait for another run if the build failed in the first step.

Reply:
Right, I used "reinstall" as a shorthand; I should've been more clear. The page you linked suggests that all these installations are run in parallel anyways (and since we're a public repo, I think we don't need to worry about paying for compute power?), so it seems my previous concern doesn't apply.

I believe Travis builds would run all tests even if a step failed; is this not the case for Actions?

Reply:
You can use `continue-on-error` to keep going, but you would have to manually check the result to see where you failed, whereas right now you can see on Discord which job failed.

More importantly, I think the three checks are semantically different and independent enough to be run as separate jobs.

Reply:
@alex-shen1 I combined `pylint` and `django` in 6e43fb2f69de7c33ff59410cc50d33324bf699e1. Does this look better? (It's definitely shorter.)

Reply:
Looks good to me!

Reply:
On second thought, the name `django` doesn't exactly convey that you're running Pylint in the job. I'm guessing that you merged those because they need Python stuff installed, so maybe the name `python` would work, but that isn't fantastic either. The other alternative I can think of is moving Pylint to `eslint` and renaming the job something like `linting`, but that sorta defeats the point of merging jobs.

Reply:
LOL I'll just split them back then.

Reply:
Reverted in 01115bab815a28416d7ddd10cca0922f0d072a3d